homeassistant/components/amcrest/config_flow.py

```diff
@@ -0,0 +1,213 @@
+            else:
+                await self.async_set_unique_id(serial_number)
+                self._abort_if_unique_id_configured(
+                    updates={CONF_HOST: user_input[CONF_HOST]}
+                )
+
+                data = {
+                    CONF_NAME: user_input[CONF_NAME],
+                    CONF_HOST: user_input[CONF_HOST],
+                    CONF_PORT: user_input[CONF_PORT],
+                    CONF_USERNAME: user_input[CONF_USERNAME],
+                    CONF_PASSWORD: user_input[CONF_PASSWORD],
+                }
+
+                # Setup options from user input to handle configuration.yaml import.
+                options = {
+                    CONF_FFMPEG_ARGUMENTS: user_input.get(
+                        CONF_FFMPEG_ARGUMENTS, DEFAULT_FFMPEG_ARGUMENTS
+                    ),
+                    CONF_RESOLUTION: user_input.get(
+                        CONF_RESOLUTION, DEFAULT_RESOLUTION
+                    ),
+                    CONF_STREAM_SOURCE: user_input.get(
+                        CONF_STREAM_SOURCE, DEFAULT_STREAM_SOURCE
+                    ),
+                    CONF_BINARY_SENSORS: user_input.get(CONF_BINARY_SENSORS, []),
+                    CONF_SENSORS: user_input.get(CONF_SENSORS, []),
+                    CONF_CONTROL_LIGHT: user_input.get(
+                        CONF_CONTROL_LIGHT, DEFAULT_CONTROL_LIGHT
+                    ),
+                }
+
+                return self.async_create_entry(
+                    title=user_input[CONF_NAME], data=data, options=options
+                )
+
+        return self.async_show_form(
+            step_id="user", data_schema=SETUP_SCHEMA, errors=errors or {}
+        )
+
+    async def async_step_import(self, import_data: dict[str, Any]) -> FlowResult:
+        """Import Amecrest config from configuration.yaml."""
+        return await self.async_step_user(import_data)
+
+
+class AmcrestOptionsFlow(OptionsFlow):
+    """Amcrest options flow."""
+
+    def __init__(self, config_entry: ConfigEntry) -> None:
+        """Initialize an Amcrest options flow."""
+        self._config_entry = config_entry
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Conifgure the camera."""
+        errors: dict[str, str] = {}
+
+        if user_input is None:
+            user_input = dict(**self._config_entry.options)
+        else:
+            binary_sensors = set(user_input.get(CONF_BINARY_SENSORS, []))
+            for exclusive_binary_sensors in EXCLUSIVE_BINARY_SENSORS:
+                if len(binary_sensors & exclusive_binary_sensors) > 1:
+                    errors["base"] = "exclusive_binary_sensors"
+
+            if not errors:
+                return self.async_create_entry(title="", data=user_input)
+
+        return self.async_show_form(
+            step_id="init",
+            errors=errors,
+            data_schema=vol.Schema(
```

Comment:
You should move the schema to a constant - it will make it easier to read.
You can then use `add_suggested_values_to_schema` to pre-fill it with the user_input.